### ISSUE TITLE
select the file type available in the launcher

### ISSUE THREAD

**Issue by 12rambau** [jupyterlab/jupyterlab#11275]
### Problem

I'm coming from SO (https://stackoverflow.com/questions/69196788/how-to-add-a-custom-file-type-in-jupyterlab-launcher?noredirect=1#comment122854950_69196788) and the comments suggested to come back here to see if I can work on a PR.

I work with jupyterlab on daily basis and I would like to customize my launcher to add extra file type. Right now I need to create a new text file and then change the extention.

for example I would like to add a yaml file here: 
![](https://i.stack.imgur.com/kJlNx.png)

PS: looking at the source code of jupyterLab I realized that this file type (yaml) is already available. I didn't find what parameter I should change to make it appear along Python and R....

### Proposed Solution

Several solutions are possible in my case: 
1. create a custom extention based on https://github.com/jtpio/jupyterlab-python-file/blob/main/setup.py to avoid any exotic file format to end up in your repository
2. create a specific PR to add for example .rst files (as they are used in sphinx builds I think it would worth the shot) based on the following PR: https://github.com/jupyterlab/jupyterlab/pull/10397
3. add my file as the previous PR but more importantly add a parameter somewhere to select from the available file format which one I want to see in the launcher (because for example I have 0 use of the R one) 

Which one would make the most sense from your perspective ?

**Comment by krassowski** [comment]
> Would this need knowledge of which file types are registered or have mime renderers available for this to work?

Good point. I don't think this is strictly necessary, especially not for the (presumably) most common use case of opening a text editor with an appropriate extension. Another - and I now think better - solution would be to make the launcher data-driven in the same way toolbars and menus are, as suggested by @bollwyvl [on gitter](https://gitter.im/jupyterlab/jupyterlab?at=6184428898c13e755016fbf1):

```json
{
  "jupyter.lab.launchers": {
      "my-launcher-id": {"command": "some-extension:cmd", "category": "Some Extension", "args": {}}
  }
}
```

and then allow users to add/disable specific launcher elements in the settings. We could even have a GUI for that both use cases (adding and removing) - displaying a list of commands is not difficult, neither is adding a text field for the category.

**Comment by TidbitSoftware** [comment]
Not trying to hijack this issue, but it seems relevant: is there a known way to add a launcher entry that runs a simple bash script _without_ launching a notebook nor browser tab for web access to a service? I would like to do this with a config file rather than having to roll my own extension. Thanks for your time and response.

**Comment by krassowski** [comment]
https://github.com/trungleduc/jupyter_app_launcher could be of interest

**Comment by TidbitSoftware** [comment]
Hi Michael, thanks for your response. Yes, I have looked at that extension and have opened an issue with the creator to see if there's a known way to use it for this purpose.
